src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -731,6 +787,67 @@ class spell_pri_holy_words : public AuraScript
     }
 };
 
+// 373427 - Inescapable Torment
+// Called by Mind Blast - 8092, Shadow Word: Death - 32379
+class spell_pri_inescapable_torment : public SpellScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo
+        ({
+            SPELL_PRIEST_INESCAPABLE_TORMENT,
+            SPELL_PRIEST_INESCAPABLE_TORMENT_EFFECT,
+            SPELL_PRIEST_MINDBENDER_SHADOW,
+            SPELL_PRIEST_MINDBENDER_DISC
+        })
+            && ValidateSpellEffect({ { SPELL_PRIEST_INESCAPABLE_TORMENT, EFFECT_1 } });
+    }
+
+    void HandleOnHitTarget(SpellEffIndex /*effIndex*/)
+    {
+        Unit* caster = GetCaster();
+        Unit* target = GetHitUnit();
+
+        if (!caster->IsPlayer() || !caster->HasAura(SPELL_PRIEST_INESCAPABLE_TORMENT))
+            return;
+
+        Optional<ObjectGuid> mindbenderGUID = caster->GetSummonGUID(PET_PRIEST_MINDBENDER);
+        if (mindbenderGUID.has_value())
+        {
+            Unit* mindbender = ObjectAccessor::GetUnit(*caster, *mindbenderGUID);
+            if (!mindbender)
+                return;
+
+            mindbender->CastSpell(target, SPELL_PRIEST_INESCAPABLE_TORMENT_EFFECT, TRIGGERED_FULL_MASK);
+
+            if (caster->ToPlayer()->GetPrimarySpecialization() == TALENT_SPEC_PRIEST_SHADOW)
+                mindbenderSummon = caster->GetAura(SPELL_PRIEST_MINDBENDER_SHADOW);
+            else
+                mindbenderSummon = caster->GetAura(SPELL_PRIEST_MINDBENDER_DISC);
+
+            // Note: Mindbender's duration increases by SPELL_PRIEST_INESCAPABLE_TORMENT's (EFFECT_1)ms.
+            if (mindbenderSummon)
+            {
+                spellInfo = sSpellMgr->AssertSpellInfo(SPELL_PRIEST_INESCAPABLE_TORMENT, GetCastDifficulty());
+
+                Milliseconds extraDuration = Seconds(spellInfo->GetEffect(EFFECT_1).CalcValue());
+
+                mindbenderSummon->SetDuration(mindbenderSummon->GetDuration() + extraDuration.count());
```

Comment:
This doesn't work, extending duration of the aura does not extend the duration of summons

Reply:
how does one increase the SPELL_EFFECT_SUMMON's duration then?

Reply:
added new function in TempSummon to increase a TempSummon's duration, I didn't create a getter since GetTimer is used for such purpose apparently.